### PR TITLE
Android SDK configuration for advertising IDs

### PR DESCRIPTION
**Pull request by ameypv-rudder**
## Description of the change

> Added documentation for `withAutoCollectAdvertId` / `autoCollectAdvertId` for Android, Unity, React Native, Cordova, and Flutter docs.

## Type of documentation
- [ ] New documentation
- [ ] Documentation update
- [ ] Hotfix

## Notion ticket link

> [Mobile SDK configuration for Advertising ID](https://www.notion.so/rudderstacks/Mobile-SDK-s-Configuration-for-Advertising-Id-8636b98c99df4873a6667374d2f1d0d5)

### CODE REVIEW

**Review by desusai7 — APPROVED** [review]

File: stream-sources/rudderstack-sdk-integration-guides/rudderstack-android-sdk/index.mdx [hunk 1]
```diff
@@ -175,6 +175,7 @@ RudderClient rudderClient = RudderClient.getInstance(
                 .withDataPlaneUrl(DATA_PLANE_URL)
                 .withTrackLifecycleEvents(true)
                 .withRecordScreenViews(true)
+                .withAutoCollectAdvertId(false)
```
Comment (desusai7) [inline stream-sources/rudderstack-sdk-integration-guides/rudderstack-android-sdk/index.mdx:178]: We've decided to remove this line right ?

File: stream-sources/rudderstack-sdk-integration-guides/rudderstack-cordova-sdk.mdx [hunk 2]
```diff
@@ -402,14 +398,24 @@ RudderClient.putAnonymousId("CustomAnonymousId");
 
 ## Advertisement ID
```
Comment (ishachopra0401) [inline stream-sources/rudderstack-sdk-integration-guides/rudderstack-cordova-sdk.mdx:399]:
```suggestion
## Setting the advertisement ID
```

File: stream-sources/rudderstack-sdk-integration-guides/rudderstack-android-sdk/index.mdx [hunk 3]
```diff
@@ -647,18 +648,35 @@ RudderClient.putDeviceToken("your_device_token");
     </TabPanels>
 </Tabs>
 
-## Advertisement ID
+## Setting the advertisement ID using `putAdvertisingId`
```
Comment (ishachopra0401) [inline stream-sources/rudderstack-sdk-integration-guides/rudderstack-android-sdk/index.mdx:651]:
```suggestion
## Setting the advertisement ID
```

File: stream-sources/rudderstack-sdk-integration-guides/rudderstack-react-native-sdk.mdx [hunk 4]
```diff
@@ -599,14 +611,25 @@ You can set the log level to one of the following values:
 5. `DEBUG`
 6. `VERBOSE`
 
-## Advertising ID
+## Setting the advertisement ID using `putAdvertisingId`
```
Comment (ishachopra0401) [inline stream-sources/rudderstack-sdk-integration-guides/rudderstack-react-native-sdk.mdx:614]:
```suggestion
## Setting the advertisement ID
```

File: stream-sources/rudderstack-sdk-integration-guides/rudderstack-cordova-sdk.mdx [hunk 5]
```diff
@@ -402,14 +398,24 @@ RudderClient.putAnonymousId("CustomAnonymousId");
...
+
+```javascript
+RudderClient.initialize(WRITE_KEY , {
+  dataPlaneUrl: DATA_PLANE_URL,
+  loglevel: RudderClient.LogLevel.VERBOSE,
+  autoCollectAdvertId: true,
+})
+```
+
+To set the advertising ID yourself, you can use the `putAdvertisingId` method as shown:
```
Comment (ishachopra0401) [inline stream-sources/rudderstack-sdk-integration-guides/rudderstack-cordova-sdk.mdx:411]:
```suggestion
To set the advertisement ID yourself, you can use the `putAdvertisingId` method as shown:
```

File: stream-sources/rudderstack-sdk-integration-guides/rudderstack-flutter-sdk/flutter-v2.mdx [hunk 6]
```diff
@@ -125,6 +125,7 @@ You can configure your client based on the following parameters by passing them
 | `sleepTimeout`          | Integer     | Minimum waiting time to flush the events to the server.          | `10 seconds`                      |
 | `configRefreshInterval` | Integer     | Fetches the config from the dashboard after this specified time.                              | `2`                                                      |
 | `trackLifecycleEvents`  | Boolean | Determines if the SDK will capture application life cycle events automatically.                   | `true`                                                   |
+| `autoCollectAdvertId` | Boolean | Determines if the SDK will collect the advertising ID. | `false` |
```
Comment (ishachopra0401) [inline stream-sources/rudderstack-sdk-integration-guides/rudderstack-flutter-sdk/flutter-v2.mdx:128]:
```suggestion
| `autoCollectAdvertId` | Boolean | Determines if the SDK will collect the advertisement ID. | `false` |
```

File: stream-sources/rudderstack-sdk-integration-guides/rudderstack-flutter-sdk/flutter-v2.mdx [hunk 7]
```diff
@@ -469,13 +470,25 @@ By default, RudderStack uses `deviceId` as `anonymousId`. You can use the follow
 rudderClient.putAnonymousId(<ANONYMOUS_ID>);
 ```
 
-## Passing your Android/iOS advertising ID
+## Setting your advertisement ID
```
Comment (ishachopra0401) [inline stream-sources/rudderstack-sdk-integration-guides/rudderstack-flutter-sdk/flutter-v2.mdx:473]:
```suggestion
## Setting the advertisement ID
```

File: stream-sources/rudderstack-sdk-integration-guides/rudderstack-flutter-sdk/index.mdx [hunk 8]
```diff
@@ -417,21 +418,35 @@ By default, RudderStack uses `deviceId` as `anonymousId`. You can use the follow
 rudderClient.putAnonymousId(<ANONYMOUS_ID>);
 ```
 
-## Passing your Android/iOS advertising ID
+## Setting your advertisement ID
```
Comment (ishachopra0401) [inline stream-sources/rudderstack-sdk-integration-guides/rudderstack-flutter-sdk/index.mdx:421]:
```suggestion
## Setting the advertisement ID
```

**Review by ishachopra0401 — APPROVED** [review]

**Pull request merged**